app/Models/User.php

```diff
@@ -540,7 +540,7 @@ public function rank()
 
     public function rankHistories()
     {
-        return $this->hasMany(RankHistory::class);
+        return $this->hasMany("App\Models\RankHistory", 'user_id', 'user_id');
```

Comment:
laravel doesn't support composite primary key so the only way is to use conditional delete on class level.

(also fwiw I tried reverting this line and ran the seeder again; it went fine)